### PR TITLE
Update integration tests

### PR DESCRIPTION
**Pull request by michaelckelly**
This pull request updates the integration tests to use `/sandbox/public_token/create` and `/item/public_token/exchange`.

### CODE REVIEW

File: test/PlaidClientTest.js [hunk 1]
```diff
@@ -550,28 +350,77 @@ describe('plaid.Client', () => {
...
-          pCl.createItem({
-            username: testConstants.USERNAME,
-            password: testConstants.PASSWORDS.GOOD
-          }, testConstants.INSTITUTION, testConstants.PRODUCTS, {
-            transactions: {
-              start_date: now,
-              end_date: now,
-              await_results: true
+
+        var getTransactionsWithRetries = (accessToken, startDate, endDate,
```
Comment (nathankot) [inline test/PlaidClientTest.js]: (nit) would be cool to use some generic retry function for both of these

File: test/PlaidClientTest.js [hunk 2]
```diff
@@ -591,8 +440,9 @@ describe('plaid.Client', () => {
           });
         });
 
-        it('all transactions (promise)', cb => {
-          pCl.getAllTransactions(accessToken, now, now).then(transactions => {
+        it.skip('all transactions (promise)', cb => {
```
Comment (nathankot) [inline test/PlaidClientTest.js]: Can we document why these are skipped?

Comment (michaelckelly) [reply test/PlaidClientTest.js]: :zap:

Opened https://github.com/plaid/plaid-node/issues/186 and referenced that in a comment

File: test/PlaidClientTest.js (hunk 1 above)
Comment (michaelckelly) [reply test/PlaidClientTest.js]: :zap: to the idea but going to punt this for the future

**Pull request merged**